gene2phenotype_project/gene2phenotype_app/serializers.py

```diff
@@ -980,111 +986,102 @@ class Meta:
-    def format_json(data):
-        data_json_input = {}
-
-        # In the HTML form the locus is a list -> ['CDH11']
-        # With raw data the locus is a string -> 'CDH11'
-        if isinstance(data.get('locus'), list):
-            data_json_input['locus'] = data.get('locus')[0]
-        else:
-            data_json_input['locus'] = data.get('locus')
+    
+    def validate(self, data):
+        """
+            Validate the input data for curation.
 
-        # allelic requirement and confidence by default are 'None'
-        # disease by default is ''
-        # This format is consistent between the HTML form and the raw data
-        check_keys = {'allelic_requirement': None, 'disease': '', 'confidence': None}
+            Args:
+                data: The data to be validated.
 
-        for key in check_keys.keys():
-            if isinstance(data.get(key), list):
-                if len(data.get(key)) > 0 and data.get(key)[0] == "":
-                    data_json_input[key] = check_keys[key]
-                else:
-                    data_json_input[key] = data.get(key)[0]
-            else:
-                data_json_input[key] = data.get(key)
-
-        check_from_list = ['publications', 'phenotypes', 'cross_cutting_modifier', 'variant_types',
-                           'variant_consequences', 'molecular_mechanism', 'panel']
+            Returns:
+                The validated data.
 
-        for tag in check_from_list:
-            if len(data.get(tag)) > 0 and isinstance(data.get(tag)[0], list):
-                data_json_input[tag] = data.get(tag)[0]
-            else:
-                data_json_input[tag] = data.get(tag)
+            Raises:
+                serializers.ValidationError: If the data is already under curation or if the user does not have permission to curate on certain panels.
+        """
 
-        return data_json_input
+        data_copy = copy.deepcopy(data) # making a copy of this so any changes to this are only to this 
+        
+        data_dict = self.convert_to_dict(data_copy)
 
-    # Check if JSON objects have the same data (only compares first layer)
-    def compare_curation_data(input_json_data, user_obj):
-        user_sessions_queryset = CurationData.objects.filter(user=user_obj.id)
-        for curation_data in user_sessions_queryset:
-            data_json = curation_data.json_data
-            if data_json == input_json_data:
-                return curation_data
+        user_email = self.context.get('user')
+        user_obj = User.objects.get(email=user_email)
         
-        return None
+        # Check if JSON is already in the table
+        curation_entry = self.compare_curation_data(data_dict, user_obj.id)
+   
+        if curation_entry: # Throw error if data is already stored in table
+            raise serializers.ValidationError({"message": f"Data already under curation. Please check session '{curation_entry.session_name}'"})
+        else:
```

Comment:
The `else` should be replace by `elif`:
`elif len(data_dict["json_data"]["panels"]) >= 1:`